pkg/util/cmdvalues/cloudproviderarray.go

```diff
@@ -0,0 +1,68 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdvalues
+
+import (
+	"fmt"
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/spf13/pflag"
+)
+
+type CloudProviderArrayValue struct {
+	allowedProvider map[v1beta1.CloudProvider]bool
+	cloudproviders  map[v1beta1.CloudProvider]bool
+	value           *[]v1beta1.CloudProvider
+	changed         bool
+}
+
+func NewCloudProviderArrayValue(value *[]v1beta1.CloudProvider, allowed ...v1beta1.CloudProvider) pflag.Value {
+	cpvalue := &CloudProviderArrayValue{
+		cloudproviders: make(map[v1beta1.CloudProvider]bool),
+		value:          value,
+		changed:        false,
+	}
+	for _, cp := range allowed {
+		cpvalue.allowedProvider[cp] = true
+	}
+	return cpvalue
+}
+
+func (v *CloudProviderArrayValue) String() string {
+	return ""
```

Comment:
because we cannot simply `strings.join(providers...)` as providers is not a string and we would need to cast every provider in the value. And as we do not need the `String()` I think this is not worthwhile implementing?